src/config/backends/extlinux.ts

```diff
@@ -0,0 +1,262 @@
+
+		return {
+			...appendConfig,
+			...fdtConfig,
+		};
+	}
+
+	public async setBootConfig(opts: ConfigOptions): Promise<void> {
+		// First get a representation of the configuration file, with all balena-supported configuration removed
+		let confContents: string;
+
+		try {
+			confContents = await fs.readFile(
+				ExtlinuxConfigBackend.bootConfigPath,
+				'utf-8',
+			);
+		} catch {
+			// In the rare case where the user might have deleted extlinux conf file between linux boot and supervisor boot
+			// We do not have any backup to fallback too; warn the user of a possible brick
+			throw new Error(
+				'Could not find extlinux file. Device is possibly bricked',
+			);
+		}
+
+		// Parse ExtlinuxFile from file contents
+		const parsedBootFile = ExtlinuxConfigBackend.parseExtlinuxFile(
+			confContents,
+		);
+
+		// Get default label to know which label entry to edit
+		const defaultLabel = ExtlinuxConfigBackend.findDefaultLabel(parsedBootFile);
+
+		// Get the label entry we will edit
+		const defaultEntry = ExtlinuxConfigBackend.getLabelEntry(
+			parsedBootFile,
+			defaultLabel,
+		);
+
+		// Set `FDT` directive if a value is provided
+		if (opts.fdt) {
+			defaultEntry.FDT = this.fdtDirective.value(opts);
+		}
+
+		// Remove unsupported options
+		const appendOptions = _.pickBy(
+			opts,
+			// supportedConfigValues has values AND directives so we must filter directives out
+			(_value, key) => this.isSupportedConfig(key) && !this.isDirective(key),
+		);
+
+		// Add config values to `APPEND` directive
+		defaultEntry.APPEND = this.appendDirective.value(
+			appendOptions,
+			defaultEntry.APPEND,
+		);
+
+		// Write new extlinux configuration
+		return await remountAndWriteAtomic(
+			ExtlinuxConfigBackend.bootConfigPath,
+			ExtlinuxConfigBackend.extlinuxFileToString(parsedBootFile),
+		);
+	}
+
+	public isSupportedConfig(configName: string): boolean {
+		return _.includes(ExtlinuxConfigBackend.supportedConfigValues, configName);
+	}
+
+	public isBootConfigVar(envVar: string): boolean {
+		return _.startsWith(envVar, ExtlinuxConfigBackend.bootConfigVarPrefix);
+	}
+
+	public processConfigVarName(envVar: string): string {
+		return envVar.replace(ExtlinuxConfigBackend.bootConfigVarRegex, '$2');
+	}
+
+	public processConfigVarValue(_key: string, value: string): string {
+		return value;
+	}
+
+	public createConfigVarName(configName: string): string {
+		return `${ExtlinuxConfigBackend.bootConfigVarPrefix}${configName}`;
+	}
+
+	private isDirective(configName: string): boolean {
+		return _.includes(
+			ExtlinuxConfigBackend.supportedDirectives,
+			configName.toUpperCase(),
+		);
+	}
+
+	private static parseExtlinuxFile(confStr: string): ExtlinuxFile {
+		const file: ExtlinuxFile = {
+			globals: {},
+			labels: {},
+		};
+
+		// Firstly split by line and filter any comments and empty lines
+		let lines = confStr.split(/\r?\n/);
+		lines = _.filter(lines, (l) => {
+			const trimmed = _.trimStart(l);
```

Comment:
You could technically include this trim in the split above with
```ts
let lines = confStr.split(/\r?\n\s*/);
```
I'm not sure I'm a big fan of it, but I found it interesting to note.

And actually I just realized you could replace the entire filter with
```ts
let lines = confStr.split(/(?:\r?\n[\s#]*)+/);
```

Reply:
Yup this works great!